packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -250,11 +275,13 @@ Plugin _pluginFromPubspec(String name, Uri packageRoot) {
     return null;
   }
   final String packageRootPath = fs.path.fromUri(packageRoot);
+  final YamlMap dependencies = pubspec['dependencies'];
   printTrace('Found plugin $name at $packageRootPath');
   return Plugin.fromYaml(
     name,
     packageRootPath,
     flutterConfig['plugin'] as YamlMap,
+    dependencies == null ? <String>[] : <String>[...dependencies?.keys],
```

Comment:
Since you've already checked if `dependencies` is null
```suggestion
    dependencies == null ? <String>[] : dependencies.keys.toList(),
```